app.py

```diff
@@ -1,34 +1,32 @@
+from auth import auth_bp
+from user import user_bp
+from decorators import *
+from db import db, User
+from flask import Flask, render_template, redirect, session, request, url_for, make_response, flash
+from flask_sqlalchemy import SQLAlchemy
 import json
-from functools import wraps
-from flask import Flask, render_template, redirect, session
 
-import auth
-from auth import auth_bp
 
 app = Flask(__name__)
-
 app.config.from_file("config.json", load=json.load)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///ictm-teaching.db'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
 
 # Core blueprints
 app.register_blueprint(auth_bp, url_prefix="/auth")
+app.register_blueprint(user_bp, url_prefix="/user")
 
-
-# Decorators
-def login_required(f):
-    @wraps(f)
-    def decorated_function(*args, **kwargs):
-        if not session.get("logged_in", False):
-            return auth.login()
-        else:
-            return f(*args, **kwargs)
-    return decorated_function
-
+db.init_app(app)
+with app.app_context():
```

Comment:
Watch out, it may be more careful to create a standalone script to create the database. I guess that in the most optimistic scenario, SQLAlchemy issues `IF NOT EXISTS` SQL commands but if your database schema changes it may create the new tables without updating the older, resulting in a broken database.